src/main/java/com/chaosthedude/naturescompass/util/BiomeSearchWorker.java

```diff
@@ -70,14 +77,25 @@ public boolean hasWork() {
 	@Override
 	public boolean doWork() {
 		if (hasWork()) {
-			if (direction == Direction.NORTH) {
-				z -= sampleSpace;
-			} else if (direction == Direction.EAST) {
-				x += sampleSpace;
-			} else if (direction == Direction.SOUTH) {
-				z += sampleSpace;
-			} else if (direction == Direction.WEST) {
-				x -= sampleSpace;
+			final int step = Math.min(maxStep, sampleSpace + (int)(lastStep * sampleMomentum));
+			int stepsRemaining = step;
+			while (stepsRemaining > 0) {
+				final int unseenLength = nextLength - length;
+				final int segment = Math.min(unseenLength, stepsRemaining);
```

Comment:
This section with `step`, `stepsRemaining`, etc. is a bit hard to understand at a glance. Might be worth using more descriptive variable names and some comments if necessary.

Reply:
Ah yeah, made this a bit clearer. Doing this because I saw some weird sampling clusters around corners; see the plots below.

![stepboundaries](https://user-images.githubusercontent.com/1237500/127969454-08515137-72fb-4170-82b7-988977f81d2a.PNG)